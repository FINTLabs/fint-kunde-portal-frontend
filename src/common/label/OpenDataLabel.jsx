import React from "react";
import { Chip, Tooltip} from "@mui/material";
import { withStyles } from '@mui/styles';


const styles = theme => ({
    chip: {
        margin: theme.spacing(0.5),
    },
});

function OpenDataLabel(props) {
    const {classes} = props;
    return (
        <Tooltip
            placement="top"
            title="Dette er åpne data som ikke trenger å tilordnes organisasjonen. Denne komponenten trenger ikke adapter eller klienter."
        >
            <Chip
                size="small"
                variant="outlined"
                label="Åpne Data"
                className={classes.chip}
            />
        </Tooltip>
    );
}

export default withStyles(styles)(OpenDataLabel);
