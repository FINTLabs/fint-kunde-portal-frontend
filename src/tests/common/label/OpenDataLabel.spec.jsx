import React from "react";
import { create } from "react-test-renderer";
import OpenDataLabel from "../../../common/label/OpenDataLabel";

describe("Feature component", () => {
  const component = create(<OpenDataLabel />);
  it("should match snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
