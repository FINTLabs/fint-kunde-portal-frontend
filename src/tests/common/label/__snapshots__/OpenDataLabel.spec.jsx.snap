// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature component should match snapshot 1`] = `
<div
  aria-describedby={null}
  className="MuiChip-root OpenDataLabel-chip-1 MuiChip-colorPrimary"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Dette er åpne data som ikke trenger å tilordnes organisasjonen. Denne komponenten trenger ikke adapter eller klienter."
>
  <div
    className="MuiAvatar-root MuiChip-avatar OpenDataLabel-avatar-2 MuiChip-avatarColorPrimary MuiAvatar-colorDefault"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiChip-avatarChildren"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6h1.9c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm0 12H6V10h12v10z"
      />
    </svg>
  </div>
  <span
    className="MuiChip-label"
  >
    Åpne Data
  </span>
</div>
`;
